src/python/WMComponent/DBS3Buffer/MySQL/NewSubscription.py

```diff
@@ -39,6 +39,8 @@ def _createPhEDExSubBinds(self, datasetID, subscriptionInfo, custodialFlag):
 
         # DeleteFromSource is not supported for move subscriptions
         delete_blocks = None
+        # if None, force it to 0 as per the table schema
+        dataLifetime = subscriptionInfo.get('DatasetLifetime', 0) or 0
```

Comment:
This line is quite ambiguous, I think. The `.get()` method is returning a proper default value, but we'd better have an explicit check for `None` value.

Reply:
Thanks @vkuznet 
I do see what the line is doing. I did not mean it is wrong. I simply wanted to say: The reader may just miss the fact that there is also another possibility the value to be None at first place - provided as such value from the dictionary itself. But as written here also works  perfectly fine. Lets not waste time on such minor things. Forget about my comment @amaltaro.  The code is good.